Catalog.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Net.Http.Json;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+namespace Lekcia11_cvicenie
+
+{
+    internal class Catalog
+    {
+        private List<Product> _products;
+
+        public Catalog() 
+        { 
+            _products = new List<Product>();
+        }
+
+        public void AddProduct(Product product)
+        {
+                _products.Add(product);
+        }
+
+        public List<string> SerializeProducts()
+        {
+            return _products.Select(product => JsonSerializer.Serialize(product)).ToList();
+        }
+
+        public void DeserializeProducts(List<string> jsonProducts)
+        {
+            foreach (var json in jsonProducts)
+            {
+                try
+                {
+                    var product = JsonSerializer.Deserialize<Product>(json);
+                    if (product != null)
+                    {
+                        _products.Add(product);
+                    }
+                }
+                catch (JsonException e)
+                {
+                    Console.WriteLine($"Chyba pri deserializaci: {e.Message}");
```

Comment:
Tady (a i v tom dalsim catch bloku) by stalo za to zvazit, zda vyjimku jen chytnou a provest vypis do konzole, nebo zda provest vypis do konzole a vyjimku vyhodit i vyse
```suggestion
                    Console.WriteLine($"Chyba pri deserializaci: {e.Message}");
                    throw e;
```

Reply:
Necham to na tobe, ale chtel jsem to zminit z toho duvodu, aby sem ti ukazal, ze tenhle pattern (chytil jsem vyjimku -> neco na zaklade ni udelam -> vyhodim ji dal) muze byt uzitecny a i se casto pouziva.

Oba dva pristupy (vyhodit vys / nevyhodit) jsou ok, ale zkus se mrknout na rozdil v rozdilu chovani cele aplikace.

Tento pristup umozni "ignorovat" chybu, ktera nastala pro jeden prvek listu. Pristup s vyhozenim vyjimky dal ukonci zpracovani metody a tedy pri tom, kdyz narazi na spatny produkt, tak uz nezpracovava ani ty dalsi (muze byt zadouci) :)